src/crosswordgenerator.ts

```diff
@@ -16,138 +17,164 @@ const emptyTile: tileCrossWord = {
+        while(answers.length>0 && tries < 50) {
             //choose random word
-            
             let indexOfCurrentAnswer = Math.floor(Math.random() * answers.length)
             let currentAnswer = answers[indexOfCurrentAnswer];
+            tries++;
             let placedWord = await tryPlaceWord(currentAnswer,crossword);
             if(placedWord){
-                answers.splice(indexOfCurrentAnswer,1)
+                answers.splice(indexOfCurrentAnswer,1);
+                tries = 0;
             }
         }
-
         resolve(crossword);
     })
 }
 
+async function removeWord(word: answer, crossword:tileCrossWord[][], questions: question[]){
+    let questIndex = -1;
+    questions.forEach((quest, i) => {
+        if(quest.answer == word.answer){
+            questIndex = i;
+        }
+    })
+    startpoints.forEach((tile) => {
+        if(Number(tile.answer)>questIndex+1){
+            tile.answer = String(Number(tile.answer)-1);
+        }
+    })
+    questions.splice(questIndex,1);
+}
+
 async function tryPlaceWord(word: answer, crossword:tileCrossWord[][]):Promise<boolean>{
     return new Promise(async (resolve) => {
-        let placed = false;
         let intersections:position[] = new Array();
         let currentChar:String = '';
         await getIntersections(word, crossword, currentChar, intersections);
         intersections.forEach(async (inter,index) => {
-            let intersect = await checkIntersection(word,inter, crossword);
+            let intersect = await checkIntersection(word, inter, crossword);
             if(intersect){
                 intersections.splice(index,1)
             }
         })
-        while(intersections.length > 0){
+        let placed = false;
+        while(intersections.length>0 && !placed){
             let indexOfCurrentIntersection = Math.floor(Math.random() * intersections.length);
             let curInter = intersections[indexOfCurrentIntersection];
+            intersections.splice(indexOfCurrentIntersection,1);
             let fit = true;
-            for(let i = 0;i<word.answer.length; i++){
-                if(curInter.x-curInter.positionInWord+i>=0 && curInter.x-curInter.positionInWord+i<=rows){ 
+            const characters = word.answer.split("");
+            characters.forEach(async (c,i)=>{
+                if(curInter.x-curInter.positionInWord+i>=0 && curInter.x-curInter.positionInWord+i<rows){
                     if(word.answer[i]!=crossword[curInter.x-curInter.positionInWord+i][curInter.y].answer){
                         if(crossword[curInter.x-curInter.positionInWord+i][curInter.y].answer!="empty"){
                             fit = false;
-                            break;
                         }
                     }
                 }
-            }
+            })
             if(curInter.x-curInter.positionInWord-1>=0){
                 if(crossword[curInter.x-curInter.positionInWord-1][curInter.y].answer!="empty"){
```

Comment:
Do I see correctly that it is currently absolutely impossible to have the answer `empty` with this approach?